packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -143,6 +143,9 @@ class BlockListBlock extends Component {
 					styles.dashedBorderColor,
 					styles.dashedBorderColorDark
 				),
+				...( ! hasChildren &&
+					this.props.name === 'core/column' &&
```

Comment:
I think we also need to figure out something better than calling specific name in here. Maybe we should create something similar to `getGroupingBlockName()` selector for `Group` block

Reply:
From what I've understood so far, this check is preventing the _child border_ to show in an empty column, because it will be showing an appender button:

![Simulator Screen Shot - iPhone 11 - 2020-02-11 at 18 11 03](https://user-images.githubusercontent.com/8739/74260665-3c92a580-4cfa-11ea-8724-36b76f12489a.png)

Without this, a selected `columns` that had an empty column would show a dashed border around the appender button:

![Simulator Screen Shot - iPhone 11 - 2020-02-11 at 18 13 08](https://user-images.githubusercontent.com/8739/74260731-57fdb080-4cfa-11ea-9190-319f39e66477.png)

This differs from the group block, as group only shows the appender when the group itself is selected. When the parent is selected, it shows the child border and the placeholder inside:

![Simulator Screen Shot - iPhone 11 - 2020-02-11 at 18 16 08](https://user-images.githubusercontent.com/8739/74260907-9eeba600-4cfa-11ea-87d8-7800aea61eb1.png)

I think in this case the code complexity comes from complexity in the UX. I don't think it's a good exception to the rule for the child borders, and I think the second screenshot actually looks more consistent. Without the extra border, the appender might look more like an option to add a new column than to add content to an existing one:

![Simulator Screen Shot - iPhone 11 - 2020-02-11 at 18 19 42](https://user-images.githubusercontent.com/8739/74261214-2cc79100-4cfb-11ea-808f-3f7eae1a454d.png)

@iamthomasbishop @pinarol any thoughts about revisiting this one?

Reply:
First of all thanks for the analysis @koke this explains a lot.

Speaking of of borders, if column is there I think we should show the dashed borders of it, otherwise it looks like column isn't there yet and it can be misleading. Plus, it brings a consistency into code and makes it easier to maintain.

Same thing goes for the appender, it would be really good to agree on a single principle that every block can use. I don't have a strong preference about which one to choose, but as soon as we are consistent it should be ok.

In general, I'd really prefer avoiding diverging the inner blocks behavior between different blocks unless it is very necessary. If we need checks like "if this is inside a Column do this, if inside a Group do that, if media & text then do the other thing" in the `block-list` code, I think it indicates [code smell](https://en.wikipedia.org/wiki/Code_smell) and it should be revisited, sometimes the solution is in the code alone but other times it may show us the complexity in the UX.

Reply:
@pinarol @koke 
>Speaking of borders, if column is there I think we should show the dashed borders of it, otherwise it looks like column isn't there yet and it can be misleading. Plus, it brings a consistency into code and makes it easier to maintain.

That was the original approach because it comes automatically with bordering style already implemented for a `InnerBLock` and do not need any extra changes in block list. However after discussion with @iamthomasbishop [here](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1430#issuecomment-576855897) we decide to give a try to remove dashed border for Column placeholder. 

I'm not sure what solution we can go with but if we want to being constant without implement hacky solution inside block list I think we should go with the original one as @koke proposed.

Another thing is that during work on [this PR](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1789) I talked with @koke about showing the Group placeholder in nested structures because it also duplicate the borders (the conclusion was that we also want to not show it). This case I think is related if we would like to find common consistent approach.

There is one more think that I found when I thinking about this. In the Group we render the placeholder on not selected Group and the Appender when Group is selected.

In `Columns` we render the `Column`, so `Columns` is only the container for `Column` blocks. shouldn't we render the  'placeholder' for each `Column` block when the `Columns` block is selected and show the `Appender` only after select one of the `Column `?

Reply:
> ... I think it indicates code smell and should be revisited, sometimes the solution is in the code alone but other times it may show us the complexity in the UX.

With my latest update on sample branch that I've open [here](https://github.com/WordPress/gutenberg/compare/rnmobile/try/column-block-alignment). The `name==='core/column'` check is applied in 3 place:

1. to set alternate placeholder style - if we decide to revert placeholder to original with dashed border this one will be gone
2. to adjust verticalMargin of Column block 
3. to set flex on Block container holding BlockEdit ( this make me able to stretch the content and make vertical alignment)

The

Reply:
@iamthomasbishop could you check @koke 's [message](https://github.com/WordPress/gutenberg/pull/19013/files#r377781183) and see if we can unify the UX on those topics?

Reply:
@iamthomasbishop 

This is how the latest version works ( I changes the placeholder behaviour according to above @koke comment. It has he dashed border )

<img src="https://user-images.githubusercontent.com/21242757/74648109-4d6d6c00-517d-11ea-8885-e81a4d6b2485.gif" width="300" />

@pinarol 
I have also fixed "bouncing" and allow to show placeholder side-by side (point 1 and 5) according to [this list](https://github.com/WordPress/gutenberg/pull/19013#issuecomment-584203520)